tests/unit/lms/resources/_js_config_test.py

```diff
@@ -0,0 +1,161 @@
+import datetime
+from unittest import mock
+
+import jwt
+import pytest
+from pyramid.httpexceptions import HTTPBadRequest
+
+from lms.resources import LTILaunchResource
+from lms.resources._js_config import JSConfig
+from lms.values import HUser
+
+
+class TestJSConfig:
+    """General unit tests for JSConfig."""
+
+    def test_it_is_mutable(self, config):
+        config.update({"a_key": "a_value"})
+
+        assert config["a_key"] == "a_value"
+
+    @pytest.mark.parametrize(
+        "context_property", ["provisioning_enabled", "h_user", "h_groupid"]
+    )
+    def test_it_raises_if_a_context_property_raises(
+        self, context, context_property, pyramid_request
+    ):
+        # Make reading context.<context_property> raise HTTPBadRequest.
+        setattr(
+            type(context),
+            context_property,
+            mock.PropertyMock(side_effect=HTTPBadRequest("example error message")),
+        )
+
+        with pytest.raises(HTTPBadRequest, match="example error message"):
+            JSConfig(context, pyramid_request).config
+
+
+class TestJSConfigAuthToken:
+    """Unit tests for the "authToken" sub-dict of JSConfig."""
+
+    def test_it(
+        self, authToken, bearer_token_schema, BearerTokenSchema, pyramid_request
+    ):
+        BearerTokenSchema.assert_called_once_with(pyramid_request)
+        bearer_token_schema.authorization_param.assert_called_once_with(
+            pyramid_request.lti_user
+        )
+        assert authToken == bearer_token_schema.authorization_param.return_value
+
+    def test_it_is_None_for_non_lti_users(
+        self, authToken, BearerTokenSchema, context, pyramid_request
+    ):
+        pyramid_request.lti_user = None
+
+        assert JSConfig(context, pyramid_request).config["authToken"] is None
+
+    @pytest.fixture
+    def authToken(self, config):
+        return config["authToken"]
```

Comment:
I think this might be better inline. We only use it once, and it adds a layer of indirection about what the test is asserting.

Reply:
If we're keeping this for future tests + consistency, then this function should probably be called `config()` as it is in the other classes.